gh-116175

a1448125 opened the pull request:
[https://github.com/python/cpython/issues/116175](url)

When parsing a date string without a year and the date is February 29th, there is a 'day is out of range for month‘ issue.

1900 is not a leap year.






<!-- gh-issue-number: gh-116175 -->
* Issue: gh-116175
<!-- /gh-issue-number -->

a1448125 committed:
When parsing a date string without a year and the date is February 29th, there is a 'day is out of range for month' issue.

pganssle commented on Lib/_strptime.py:
```diff
@@ -536,7 +536,7 @@ def _strptime(data_string, format="%a %b %d %H:%M:%S %Y"):
         # the caller didn't supply a year but asked for Feb 29th. We couldn't
         # use the default of 1900 for computations. We set it back to ensure
         # that February 29th is smaller than March 1st.
-        year = 1900
```
This is not the right way to fix this. If anything you should just remove the code, but clearly it's here for a reason, so we should discuss why it's here and decide on a course of action in the bug.

a1448125 committed:
Update _strptime.py

Deleting this judgment might be a better way to solve the matching rule for February 29th in leap years.
